ksql-engine/src/main/java/io/confluent/ksql/planner/plan/KsqlStructuredDataOutputNode.java

```diff
@@ -107,13 +110,6 @@ public Field getKeyField() {
       addAvroSchemaToResultTopic(outputNodeBuilder);
     }
 
-    final int partitions = (Integer) outputProperties.getOrDefault(
-        KsqlConfig.SINK_NUMBER_OF_PARTITIONS_PROPERTY,
-        ksqlConfig.getInt(KsqlConfig.SINK_NUMBER_OF_PARTITIONS_PROPERTY));
```

Comment:
Is `SINK_NUMBER_OF_PARTITIONS_PROPERTY` not used at all?  Should we remove it / deprecate it?  Probably need to call this out in release notes.

Reply:
I'm more suggesting that it shouldn't be being used any more if we're now getting the partition count and rf from the source.

Reply:
Now that I added the legacy support I need to keep them.